hudi-common/src/main/java/org/apache/hudi/common/table/timeline/HoodieDefaultTimeline.java

```diff
@@ -203,6 +204,12 @@ public HoodieDefaultTimeline findInstantsInRange(String startTs, String endTs) {
         getInstantsAsStream().filter(s -> HoodieTimeline.isInRange(s.getTimestamp(), startTs, endTs)), details);
   }
 
+  @Override
+  public HoodieDefaultTimeline findInstantsInClosedRange(String startTs, String endTs) {
```

Comment:
I don't see this function being called from anywhere? Also, wondering if this time interval can be made half open? There are advantages in having time intervals as half open instead of closed as described [here](https://wrschneider.github.io/2014/01/07/time-intervals-and-other-ranges-should.html) and [here](https://stackoverflow.com/questions/9795391/is-there-a-standard-for-inclusive-exclusive-ends-of-time-intervals).

Reply:
I agree on half open range, but we need this functionality to be able to handle batches within the same commit. If its made half open it will change the behavior. This function is called from onehouse-dataplane repo

Reply:
Its used in s3 incr job estimators